### ISSUE TITLE
Compiler crashes with ternary operator on matrices with 2021 HLSL enabled

### ISSUE THREAD

**Issue by xander890** [microsoft/DirectXShaderCompiler#4628]
The newest released version of the compiler (July 2022) crashes on the ternary select operator ?: on matrices, when the -HV 2021 argument is passed. 

We know this needs to be replaced by the new select() keyword, but this bug really hinders transitions from legacy code, since it's very hard to spot where the error actually is. Also, this is bad for people accidentally coding in the old convention, since they don't get a compile error, just a crash on a potentially 3000+ lines shader file.

Would it be possible to fix it and generate a compile error with line information?

Minimal repro (target ps_6_0, options -O3 -HV 2021):

```
struct PSInput
{
    float4 color : COLOR;
    int dynamic : DYN0;
};

struct tests
{
    float3x4 test0;   
    float3x4 test1;   
};

cbuffer constbuffer : register(b0)
{
    tests Element0;
}
    

float4 PSMain(PSInput input) : SV_TARGET
{
    float3x4 m = input.dynamic == 0? Element0.test0 : Element0.test1;
    return float4(mul(m, input.color), 1.0f);
}
```

**Comment by bioglaze** [comment]
This looks similar to a bug I reported earlier: https://github.com/microsoft/DirectXShaderCompiler/issues/4434

**Comment by pow2clk** [comment]
Confirmed that the pending fix for #4434 will fix this issue too. We've renewed our attention to that fix. In the meantime, I'll close this as a dupe.
